ariadne/src/Ariadne/Config/Cardano.hs

```diff
@@ -2,75 +2,336 @@ module Ariadne.Config.Cardano
+    , Cardano.npEnableMetrics = npEnableMetrics
+    , Cardano.npEkgParams = npEkgParams
+    , Cardano.npStatsdParams = Nothing
+    , Cardano.npNetworkConfig = npNetworkConfig
+    , Cardano.npBehaviorConfig = npBehaviorConfig
+    }
+
+data NetworkConfigOpts = NetworkConfigOpts
+    { ncoTopology :: !(Presence Topology)
+    , ncoPort  :: !Word16
+    } deriving Show
+
+instance Eq NetworkConfigOpts where
+    nco == nco' = ncoPort nco ==  ncoPort nco'
+
+toCardanoNetworkConfigOpts :: NetworkConfigOpts -> Cardano.NetworkConfigOpts
+toCardanoNetworkConfigOpts NetworkConfigOpts{..} = Cardano.NetworkConfigOpts
+    { Cardano.ncoTopology = ncoTopology ^? _File
+    , Cardano.ncoKademlia = Nothing
+    , Cardano.ncoSelf = Nothing
+    , Cardano.ncoPort = ncoPort
+    , Cardano.ncoPolicies = Nothing
+    , Cardano.ncoBindAddress = Nothing
+    , Cardano.ncoExternalAddress = Nothing
+    }
+
+data ConfigurationOptions = ConfigurationOptions
+    { cfo :: !(Presence Configuration) }
+    deriving (Eq, Show)
+
+instance Default ConfigurationOptions where
+    def = ConfigurationOptions $ There defConfiguration
+
+toCardanoConfigurationOptions :: ConfigurationOptions -> Cardano.ConfigurationOptions
+toCardanoConfigurationOptions ConfigurationOptions{..} = Cardano.ConfigurationOptions
+    { Cardano.cfoFilePath = maybe "" id $ cfo ^? _File
+    , Cardano.cfoKey = "mainnet_full"
+    , Cardano.cfoSystemStart = Nothing
+    , Cardano.cfoSeed = Nothing
+    }
+
+data CommonArgs = CommonArgs
+    { logConfig            :: !(Presence LoggerConfig)
+    , logPrefix            :: !(Maybe FilePath)
+    , configurationOptions :: !ConfigurationOptions
+    }
+
+-- Checked all fields except logConfig
+instance Eq CommonArgs where
+    ca == ca' = logPrefix ca == logPrefix ca'
+        && configurationOptions ca == configurationOptions ca'
+
+instance Show CommonArgs where
+    show CommonArgs{..} = "CommonArgs { " <> show logPrefix <>
+        ", " <> show configurationOptions <> " }"
+
+toCardanoCommonArgs :: CommonArgs -> Cardano.CommonArgs
+toCardanoCommonArgs CommonArgs{..} = Cardano.CommonArgs
+    { Cardano.logConfig = logConfig ^? _File
+    , Cardano.logPrefix = Nothing
+    , Cardano.reportServers = []
+    , Cardano.updateServers = []
+    , Cardano.configurationOptions = toCardanoConfigurationOptions configurationOptions
+    }
+
+newtype CardanoConfig = CardanoConfig
+    { getCardanoConfig :: CommonNodeArgs }
+    deriving (Eq, Show)
+
+instance D.Interpret CardanoConfig where
+    autoWith _ = CardanoConfig <$> interpretCommonNodeArgs
+
+instance D.Inject CardanoConfig where
+    injectWith _ = contramap getCardanoConfig injectCommonNodeArgs
+
+defTopology :: Topology
+defTopology = either (error "LoggerConfig retrieving failed during parsing") id . decodeEither' $
+    $(embedFile $ (unsafePerformIO $ (\fp -> flip take fp $ (length fp) - 8 ) <$> getCurrentDirectory)
+        <> "/config/cardano/topology.yaml")
```

Comment:
It looks rather hacky and fragile. First of all, it most likely won't work on Windows, because you are using `/`. So at least `filepath` library should be used to work with paths. Second, using `unsafePerformIO` seems weird, you can use `runIO`.
I think that simpler approach is to move `config/` to `ariadne/` and embed `config/cardano/topology.yaml` and similar files. And `config/` in the root can be symlinked to `ariadne/config/`.